homeassistant/components/bmw_connected_drive/sensor.py

```diff
@@ -85,83 +75,116 @@ def convert_and_round(
         key="charging_status",
         translation_key="charging_status",
         key_class="fuel_and_battery",
-        value=lambda x, y: x.value,
+        device_class=SensorDeviceClass.ENUM,
+        options=[
+            s.value.lower()
+            for s in (
+                ChargingState.DEFAULT,
+                ChargingState.CHARGING,
+                ChargingState.ERROR,
+                ChargingState.COMPLETE,
+                ChargingState.FULLY_CHARGED,
+                ChargingState.FINISHED_FULLY_CHARGED,
+                ChargingState.FINISHED_NOT_FULL,
+                ChargingState.INVALID,
+                ChargingState.NOT_CHARGING,
+                ChargingState.PLUGGED_IN,
+                ChargingState.WAITING_FOR_CHARGING,
+                ChargingState.TARGET_REACHED,
```

Comment:
Which one is missing?

Reply:
Not sure if I follow. The only Enum not in this list is `ChargingState.UNKNOWN` as this is not a valid state and (from previous comments) should be `None`.

Reply:
I'm sorry, I thought the discussion about this with @MartinHjelmare (https://github.com/home-assistant/core/pull/110272#discussion_r1571806397) was the reason this PR was stuck.

Thats why I reverted this in https://github.com/home-assistant/core/pull/110272/commits/947c8b87dc868eb10e556f7a3f28837ca23930bc.

Reply:
Yes that was the original point of the PR, but I still believe this is a non-issue because we got it covered. I spoke Frenck today and he was like "why not"